rpc/src/tests/module/test.rs

```diff
@@ -0,0 +1,112 @@
+#![allow(clippy::inconsistent_digit_grouping)]
+
+use ckb_chain_spec::consensus::build_genesis_epoch_ext;
+use ckb_store::ChainStore;
+use ckb_test_chain_utils::always_success_consensus;
+use ckb_types::{
+    core::{Capacity, EpochNumberWithFraction},
+    prelude::Unpack,
+    utilities::DIFF_TWO,
+};
+
+use crate::tests::{setup, RpcTestRequest, RpcTestSuite};
+
+const GENESIS_EPOCH_LENGTH: u64 = 30;
+
+#[test]
+fn test_generate_epochs() {
+    let suite = setup_rpc();
+    assert_eq!(
+        get_current_epoch(&suite),
+        EpochNumberWithFraction::new(0, 20, GENESIS_EPOCH_LENGTH)
+    );
+
+    // generate 1 epoch
+    suite.rpc(&RpcTestRequest {
+        id: 42,
+        jsonrpc: "2.0".to_string(),
+        method: "generate_epochs".to_string(),
+        params: vec!["0x1".into()],
+    });
+    assert_eq!(
+        get_current_epoch(&suite),
+        EpochNumberWithFraction::new(1, 20, GENESIS_EPOCH_LENGTH)
+    );
+
+    // generate 1(0/1) epoch
+    suite.rpc(&RpcTestRequest {
+        id: 42,
+        jsonrpc: "2.0".to_string(),
+        method: "generate_epochs".to_string(),
+        params: vec!["0x10000000001".into()],
+    });
+    assert_eq!(
+        get_current_epoch(&suite),
+        EpochNumberWithFraction::new(2, 20, GENESIS_EPOCH_LENGTH)
+    );
+
+    // generate 1/2 epoch
+    suite.rpc(&RpcTestRequest {
+        id: 42,
+        jsonrpc: "2.0".to_string(),
+        method: "generate_epochs".to_string(),
+        params: vec!["0x20001000000".into()],
+    });
+    assert_eq!(
+        get_current_epoch(&suite),
+        EpochNumberWithFraction::new(3, 5, GENESIS_EPOCH_LENGTH)
+    );
+
+    // generate 3/2 epoch
+    suite.rpc(&RpcTestRequest {
+        id: 42,
+        jsonrpc: "2.0".to_string(),
+        method: "generate_epochs".to_string(),
+        params: vec!["0x20003000000".into()],
```

Comment:
How is the hardcoded string `0x20003000000` calculated?

Reply:
```rust
assert_eq!(
    Into::<ckb_jsonrpc_types::Uint64>::into(EpochNumberWithFraction::new(0, 3, 2)).to_string(),
    "0x20003000000".to_string()
);
```

I'll try to add it in the comments, maybe other users will be confused as well.